DuckDuckGo/PrivacyStats/PrivacyStatsDatabase.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  PrivacyStatsDatabase.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import CoreData
+import PixelKit
+import PrivacyStats
+import Persistence
+
+public final class PrivacyStatsDatabase: PrivacyStatsDatabaseProviding {
+
+    public let db: CoreDataDatabase
+
+    init(db: CoreDataDatabase = make(location: URL.sandboxApplicationSupportURL)) {
+        self.db = db
+    }
+
+    public static func make(location: URL) -> CoreDataDatabase {
+        let bundle = PrivacyStats.bundle
+        guard let model = CoreDataDatabase.loadModel(from: bundle, named: "PrivacyStats") else {
+            fatalError("Failed to load model")
```

Comment:
Should this be a `fatalError`? Could we gracefully handle privacy stats not working as expected?

Reply:
Maybe we could throw an error and handle it?

Reply:
Yep, I've clearly taken the shortcut here. All other database stacks use fatalError here but it's a weak justification, as users could probably live without privacy stats, rather than without Bookmarks. Let me try to do something about it.

Reply:
Ok, now I know what's up. I've been here already, trying to make it not crash with a fatal error, but unfortunately this initialization is asynchronous and while we have the database object ready, we can only learn that it's unusable in an asynchronous callback. Reworking privacy stats to be asynchronous would require major changes to entire NTP loading mechanism. I will add it to the project to consider adding privacy stats module to NTP after a delay, but given it requires extra work I'd skip it for now.

After all, if the database crashes here, it's either:
* the user doesn't have space on disk, which is recoverable and they should take care of it anyway
* we messed up database migration which should really be caught earlier during the release process (and also doesn't have a chance of happening for the v1 ;))

Reply:
Reasonable justification.